src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Policy/Edit-AzrecoveryServicesBackupRetentionPolicyClientObject.ps1

```diff
@@ -639,7 +667,7 @@ function Edit-AzrecoveryServicesBackupRetentionPolicyClientObject {
           {
               $FullBackupPolicy =  $policyObject.SubProtectionPolicy | Where-Object { $_.PolicyType -match "Full" }
               $Index = $policyObject.SubProtectionPolicy.IndexOf($FullBackupPolicy)
-              if($policyObject.SubProtectionPolicy[$Index].SchedulePolicy.ScheduleRunFrequency -eq "Weekly")
+              if($policyObject.SubProtectionPolicy[$Index].SchedulePolicy.ScheduleRunFrequency -eq "Weekly" -and -not($EnableDailyRetention) -and ($policyObject.SubProtectionPolicy[$Index].RetentionPolicy.DailySchedule.RetentionDuration.Count -ne $null) -and ($policyObject.SubProtectionPolicy[$Index].RetentionPolicy.DailySchedule.RetentionDuration.Count -ne 0))
               {
                   $policyObject.SubProtectionPolicy[$Index].RetentionPolicy.DailySchedule = $null
                   Write-Host "Daily Retention is disabled when Schedule run frequency is weekly"
```

Comment:
Write-Warning "Daily Retention has been disabled since Schedule run frequency is weekly"